oioswift/common/middleware/container_hierarchy.py

```diff
@@ -603,6 +607,47 @@ def update_copy_headers(self, req, env2):
             req.headers['Oio-Copy-From'] = '/' + c_container + '/' + c_obj
             env2['HTTP_OIO_COPY_FROM'] = '/' + c_container + '/' + c_obj
 
+    def _handle_dir_object(self, req, start_response, account, container,
+                           path, missing=False):
+        if req.method in ('GET', 'HEAD'):
+            is_available = False
+            if not missing:
+                key = self.key(account, container, OBJ, path)
+                if self.redis_keys_format == REDIS_KEYS_FORMAT_V1:
+                    is_available = self.conn.exists(key + "/")
+                else:
+                    is_available = self.conn.hexists(key, path + "/")
+
+            if is_available:
+                # add a fake AllUsers permissions, otherwise it will
+                # be dropped by ACL checks in S3 Layer
+                oheaders = {
+                    'Content-Length': 0,
+                    'Etag': 'd41d8cd98f00b204e9800998ecf8427e',
+                    'x-object-sysmeta-swift3-acl': FAKE_ACL}
+                start_response("200 OK", oheaders.items())
+            else:
+                start_response("404 Not Found", {}.items())
```

Comment:
`{}.items()` is just an empty list.

Reply:
Done

Reply:
I said a **list**. I'm not sure sending a tuple is a good idea.